File: exercises/exercism/ruby/2019/difference_of_squares.rb
# Difference Of Squares from https://exercism.io

class Squares
  def initialize(number)
    @from = 1
    @to = number
  end

  def square_of_sum
    (from..to).sum ** 2
  end

  def sum_of_squares
    (from..to).sum { |n| n ** 2 }
  end

  def difference
    square_of_sum - sum_of_squares
  end

  private

    attr_reader :from, :to
end

# alias_method Integer#**, Integer#pow

File: practices/exercism/c/2020/difference_of_squares.c
/* Difference Of Squares from https://exercism.io */

#include "difference_of_squares.h"
#include <math.h>

unsigned int square_of_sum(unsigned int number)
{
  int result = 0;

  for (int i = 1; i <= (int)number; i++) {
    result += i;
  }

  return (int)pow(result, 2);
}

unsigned int sum_of_squares(unsigned int number)
{
  int result = 0;

  for (int i = 1; i <= (int)number; i++) {
    result += (int)pow(i, 2);
  }

  return result;
}

unsigned int difference_of_squares(unsigned int number)
{
  int result = square_of_sum(number) - sum_of_squares(number);

  return result;
}
